server/controller/common/signUp.js

```diff
@@ -0,0 +1,44 @@
+const { hash } = require('bcrypt');
+
+const { signupUser, checkEmail } = require('../../database/queries');
+const { boomify, signupValidation, signToken } = require('../../utilis');
+
+const signUp = async (req, res, next) => {
+  try {
+    const {
+      value: { password, email },
+    } = await signupValidation.validate(req.body, {
+      abortEarly: false,
+    });
```

Comment:
when I don't insert firstName it gives me status code 500 internal server error while it should be 400.
add try-catch block to handle this error to bad request instead of internal server error